common/persistence/visibility/visibility_manager_impl.go

```diff
@@ -295,6 +296,21 @@ func (p *visibilityManagerImpl) newInternalVisibilityRequestBase(
 		return nil, err
 	}
 
+	var searchAttrs *commonpb.SearchAttributes
+	if len(request.SearchAttributes.GetIndexedFields()) > 0 {
+		// Remove any system search attribute from the map.
+		// This is necessary because the validation can supress errors when trying
+		// to set a value on a system search attribute.
+		searchAttrs = &commonpb.SearchAttributes{
+			IndexedFields: make(map[string]*commonpb.Payload),
+		}
+		for key, value := range request.SearchAttributes.IndexedFields {
+			if !searchattribute.IsSystem(key) {
+				searchAttrs.IndexedFields[key] = value
+			}
+		}
+	}
```

Comment:
shall we do this only when the feature flag is enabled?

Reply:
I think it's ok to always do this. It just copying pointers, so little overhead.
If we want to check the dynamic config, it would have to be passed to visibility manager.